PWGHF/TableProducer/candidateCreator2Prong.cxx

```diff
@@ -74,23 +87,27 @@ struct HfCandidateCreator2Prong {
   OutputObj<TH1F> hCovSVZZ{TH1F("hCovSVZZ", "2-prong candidates;ZZ element of cov. matrix of sec. vtx. position (cm^{2});entries", 100, 0., 0.2)};
   OutputObj<TH2F> hDcaXYProngs{TH2F("hDcaXYProngs", "DCAxy of 2-prong candidates;#it{p}_{T} (GeV/#it{c};#it{d}_{xy}) (#mum);entries", 100, 0., 20., 200, -500., 500.)};
   OutputObj<TH2F> hDcaZProngs{TH2F("hDcaZProngs", "DCAz of 2-prong candidates;#it{p}_{T} (GeV/#it{c};#it{d}_{z}) (#mum);entries", 100, 0., 20., 200, -500., 500.)};
+  OutputObj<TH1F> hUseKForDCAFitter{TH1F("hUseKForDCAFitter", ";Use KF or DCAFitterN;entries", 2, -0.5, 1.5)}; // 0 for KF, 1 for DCAFitterN
 
   void init(InitContext const&)
   {
-    if (doprocessPvRefit && doprocessNoPvRefit) {
-      LOGP(fatal, "Only one process function between processPvRefit and processNoPvRefit can be enabled at a time.");
+    std::array<bool, 2> doprocessDF{doprocessPvRefit, doprocessNoPvRefit};
+    std::array<bool, 2> doprocessKF{doprocessPvRefitWithKF, doprocessNoPvRefitWithKF};
+    if ((std::accumulate(doprocessDF.begin(), doprocessDF.end(), 0) + std::accumulate(doprocessKF.begin(), doprocessKF.end(), 0)) != 1) {
+      LOGP(fatal, "Only one process function can be enabled at a time.");
     }
+
     ccdb->setURL(ccdbUrl);
     ccdb->setCaching(true);
     ccdb->setLocalObjectValidityChecking();
     lut = o2::base::MatLayerCylSet::rectifyPtrFromFile(ccdb->get<o2::base::MatLayerCylSet>(ccdbPathLut));
     runNumber = 0;
   }
 
-  template <bool doPvRefit = false, typename Cand>
+  template <bool doPvRefit = false, typename Cand, typename TTracks>
   void runCreator2Prong(aod::Collisions const& collisions,
```

Comment:
The functions using the DCA fitter should be named accordingly, similar to the KF versions.